code/src/containers/tables/orders-table/components/orders-table-body/OrdersTableBody.test.tsx

```diff
@@ -37,99 +56,266 @@ describe("OrdersTableBody", () => {
+      waitFor(() => {
+        expect(mockStatusChange).not.toHaveBeenCalledWith("SHIPPED");
+      });
+    });
+
+    test("not changes order status when choose the current order status", () => {
+      renderOrdersTableBody({ order: mockOrders[1], role: ROLES.SHOP_MANAGER });
+      const statusSelect = screen.getByText("orders.statuses.canceled");
+
+      fireEvent.mouseDown(statusSelect);
+
+      const options = screen.getAllByRole("option");
+
+      expect(options).toHaveLength(1);
+
+      const desiredStatus = screen.getByRole("option", {
+        name: "orders.statuses.canceled"
+      });
+
+      expect(desiredStatus).toHaveAttribute("aria-disabled", "true");
+
+      fireEvent.click(desiredStatus);
+      waitFor(() => {
+        expect(mockStatusChange).not.toHaveBeenCalledWith("CANCELED");
+      });
+    });
+
+    test("triggers isPaid status change after click on checkbox", () => {
+      renderOrdersTableBody({ order: mockOrders[0], role: ROLES.SHOP_MANAGER });
+      const checkboxBlank = screen.getByTestId("CheckBoxOutlineBlankIcon");
+      expect(checkboxBlank).toBeInTheDocument();
+
+      const checkboxWrapper = screen
+        .getByTestId("CheckBoxOutlineBlankIcon")
+        .closest("span");
+
+      expect(checkboxWrapper).not.toHaveAttribute("aria-disabled", "true");
+
+      fireEvent.click(checkboxBlank);
 
-    fireEvent.click(checkboxBlank);
-    await waitFor(() => {
       expect(mockIsPaidChange).toHaveBeenCalledWith(true);
       expect(checkboxBlank).not.toBeInTheDocument();
+
+      const checkboxChecked = screen.getByTestId("CheckBoxIcon");
+      expect(checkboxChecked).toBeInTheDocument();
     });
 
-    const checkboxChecked = screen.getByTestId("CheckBoxIcon");
-    expect(checkboxChecked).toBeInTheDocument();
-  });
+    test("shows tooltip after hover on not checked checkbox", () => {
+      renderOrdersTableBody({ order: mockOrders[0], role: ROLES.SHOP_MANAGER });
+      const checkbox = screen.getByRole("checkbox");
 
-  test("Should show tooltip after hover on not checked checkbox", () => {
-    renderOrdersTableBody(mockOrders[0]);
-    const checkbox = screen.getByRole("checkbox");
+      fireEvent.mouseOver(checkbox);
+      waitFor(() => {
+        const tooltip = screen.getByText(/ordersTable.notpaid.tooltip/);
+        expect(tooltip).toBeInTheDocument();
+      });
+    });
 
-    fireEvent.mouseOver(checkbox);
-    waitFor(() => {
-      const tooltip = screen.getByText(/ordersTable.notpaid.tooltip/);
-      expect(tooltip).toBeInTheDocument();
+    test("not trigger isPaid status change after click on checkbox, when order is canceled", () => {
+      renderOrdersTableBody({ order: mockOrders[1], role: ROLES.SHOP_MANAGER });
+      const checkbox = screen.getByTestId("CheckBoxOutlineBlankIcon");
+
+      const checkboxWrapper = screen
+        .getByTestId("CheckBoxOutlineBlankIcon")
+        .closest("span");
+
+      expect(checkboxWrapper).toHaveAttribute("aria-disabled", "true");
```

Comment:
Why do we need to test `aria`-attribute?

Reply:
Because there we don't have MUI Disable class and expect.toBeDisable doest't work
But we need to test that checkbox becomes disable.